Disaster-Response-Platform/backend/Services/feedback_service.py

```diff
@@ -0,0 +1,61 @@
+from Models.feedback_model import Feedback
+from Database.mongo import MongoDB
+from bson.objectid import ObjectId
+
+from Services.build_API_returns import *
+
+
+# Get the needs collection using the MongoDB class
+feedback_collection = MongoDB.get_collection('feedback')
+needs_collection = MongoDB.get_collection('needs')
+resources_collection = MongoDB.get_collection('resources')
+events_collection = MongoDB.get_collection('events')
+actions_collection = MongoDB.get_collection('actions')
+
+def vote(entityType:str, entityID:str, current_user:str, voteType:str) -> bool:
+    existing_vote = feedback_collection.find_one({"entityType": entityType, "entityID": entityID})   
+    if existing_vote and existing_vote['vote']== voteType:
+        raise ValueError(f"Current user already voted")
+    elif existing_vote:
+        result = feedback_collection.update_one({"entityType": entityType, "entityID": entityID}, {"$set": {"vote": voteType}})
+        if voteType == "upvote":
+            set_upvote(entityType, entityID, 1)
+            set_downvote(entityType, entityID, -1)
+        else:
+            set_upvote(entityType, entityID, -1)
+            set_downvote(entityType, entityID, 1)
+    else:
+        insert_result = feedback_collection.insert_one({"entityType":entityType, "entityID":entityID, "username": current_user, "vote": voteType})        
+        if voteType=="upvote":
+            set_upvote(entityType, entityID, 1)
+        else:
+            set_downvote(entityType, entityID, 1)
+        
+
+def get_entity_type_collection(entity_type: str):
```

Comment:
You can get rid of this as a whole. Defining the entitytype in the model as an enumeration.
And giving the names of collections to this enumeration.

Reply:
thanks for the suggestion, I'll do it with enum